python/hail/table.py

```diff
@@ -1283,17 +1282,29 @@ def types_compatible(left, right):
             for e in exprs:
                 analyze('Table.index', e, src._row_indices)
 
-            right = self.select(**{uid: self.row})
-            uids = [Env.get_uid() for i in range(len(exprs))]
+            is_key = src.key is not None and len(exprs) == len(src.key) and all(
+                exprs[i] is src._fields[list(src.key)[i]] for i in range(len(exprs)))
+            is_interval = (len(self.key) == 1
+                           and isinstance(self.key[0].dtype, hl.tinterval)
+                           and exprs[0].dtype == self.key[0].dtype.point_type)
+
+            if not is_key:
+                uids = [Env.get_uid() for i in range(len(exprs))]
+                all_uids = uids[:]
```

Comment:
why copy `uids`? (that is what this doing right?)

Reply:
yes. If we append to `all_uids` somewhere before uids is used, then that would be a big problem. This is a way to keep the code safe.